### PR TITLE
chore: use prettier to format generated sponsors.json

### PR DESCRIPTION
**Pull request by armano2**
## PR Checklist

-   [x] Addresses an existing open issue: https://github.com/typescript-eslint/typescript-eslint/pull/5061/files#r881215504
-   [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
-   [x] Steps in [CONTRIBUTING.md](https://github.com/typescript-eslint/typescript-eslint/blob/main/CONTRIBUTING.md) were taken

## Overview

Add missing EOF `\n` to generated `sponsors.json`

### CODE REVIEW

File: tools/generate-sponsors.ts [hunk 1]
```diff
@@ -190,7 +190,7 @@ async function main(): Promise<void> {
       allSponsorsConfig,
       (_, value: unknown) => value ?? undefined,
       2,
-    ),
+    ) + '\n',
```
Comment (JoshuaKGoldberg) [inline tools/generate-sponsors.ts:193]: Heh, alternately we could import from the `prettier` package (https://prettier.io/docs/en/api.html#prettierformatsource-options)...

**Review by JoshuaKGoldberg — APPROVED** [review]
😂

File: tools/generate-sponsors.ts (hunk 1 above)
Comment (Josh-Cena) [reply tools/generate-sponsors.ts:193]: > but I think there are some cases where the former will add new lines that prettier wouldn't.

Yes, with arrays.

Comment (armano2) [reply tools/generate-sponsors.ts:193]: priettier is actually smart with json, it preserves new lines with objects, thus there is no issue here

Comment (bradzacher) [reply tools/generate-sponsors.ts:193]: Yeah this is my point - prettier will potentially reformat the JSON.

i.e. for the json blob `{"a":[1,2,3,4,5,6],"b":[1,2,3,4,5]}`,
`JSON.stringify(x, null, 2)` results in
```json
{
  "a": [
    1,
    2,
    3,
    4,
    5,
    6
  ],
  "b": [
    1,
    2,
    3,
    4,
    5
  ]
}
```
but `prettier.format` for that stringified output results in
```json
{
  "a": [1, 2, 3, 4, 5, 6],
  "b": [1, 2, 3, 4, 5]
}
```

Which is why I say that if we're adding the newline for prettier, then we should enforce the blob is just formatted with prettier at generation time to ensure it's correct!

Comment (armano2) [reply tools/generate-sponsors.ts]: ok, I applied `prettier` as a formatter

**Review by JoshuaKGoldberg — APPROVED** [review]
🧹

File: tools/generate-sponsors.ts [hunk 2]
```diff
@@ -184,14 +185,23 @@ async function main(): Promise<void> {
...
-      allSponsorsConfig,
-      (_, value: unknown) => value ?? undefined,
-      2,
-    ) + '\n',
+  fs.writeFileSync(rcPath, await stringifyObject(allSponsorsConfig));
+}
+
+async function stringifyObject(data: unknown): Promise<string> {
+  const config = await prettier.resolveConfig(
+    path.join(__dirname, '../.prettierrc.json'),
```
Comment (bradzacher) [inline tools/generate-sponsors.ts]: nitpick
```suggestion
    path.resolve(__dirname, '..', '.prettierrc.json'),
```

Comment (armano2) [reply tools/generate-sponsors.ts]: you are reviewing not latest version i changed this code

**Review by bradzacher — APPROVED** [review]

**Pull request merged**